expression/builtin_arithmetic.go

```diff
@@ -180,6 +180,11 @@ type builtinArithmeticPlusIntSig struct {
 	baseBuiltinFunc
 }
 
+func (s *builtinArithmeticPlusIntSig) Clone() builtinFunc {
+	newSig := &builtinArithmeticPlusIntSig{baseBuiltinFunc: *s.baseBuiltinFunc.Clone().(*baseBuiltinFunc)}
```

Comment:
this type assertion can be avoid if we implement a special clone function for `baseBuiltinFunc` which returns a `baseBuiltinFunc` other than a `builtinFunc`